controllers/flinkcluster_controller.go

```diff
@@ -123,6 +133,15 @@ func (handler *_FlinkClusterHandler) Reconcile(
 	} else {
 		log.Info("Desired state", "JobManager service", "nil")
 	}
+	if desiredState.jmIngress != nil {
+		log.Info("Desired state", "JobManager ingress", *desiredState.jmIngress)
+	} else {
+		var msg string
+		if observedState.cluster != nil && observedState.cluster.Spec.JobManagerSpec.Ingress != nil && config.IngressHostFormat == "" {
```

Comment:
This logic should be moved to the validating webhook, reject the create request if the format is missing. Since we haven't implemented the validating webhook, it is okay to put it here for now, but add a TODO comment here please.